### ISSUE TITLE
Non-exhaustive patterns [E0004] for field shorthand field resolved to constant is not detected

### ISSUE THREAD

**Issue by Undin** [intellij-rust/intellij-rust#9363]
## Environment

* **IntelliJ Rust plugin version:** 0.4.179.4859-222-nightly
* **Rust toolchain version:** 1.63.0 (4b91a6ea7 2022-08-08) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2022.2 Ultimate Edition (IU-222.3345.118)
* **Operating system:** macOS 12.5.1
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.emulate.terminal, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
For the following code, the plugin doesn't detect [E0004](https://doc.rust-lang.org/error-index.html#E0004).

```rust
struct Point { X: i32, Y: i32 }

pub const X: i32 = 0;
pub const Y: i32 = 0;

fn func(foo: Point) {
    match foo {
        Point { X, Y } => {},
    }
}
```
It seems the corresponding analysis doesn't take into account that "variables" in struct literal patterns can refer to other items.

At the same time compiler produces
```
error[E0004]: non-exhaustive patterns: `Point { X: i32::MIN..=-1_i32, .. }` and `Point { X: 1_i32..=i32::MAX, .. }` not covered
 --> src/main.rs:7:11
  |
7 |     match foo {
  |           ^^^ patterns `Point { X: i32::MIN..=-1_i32, .. }` and `Point { X: 1_i32..=i32::MAX, .. }` not covered
  |
note: `Point` defined here
 --> src/main.rs:1:8
  |
1 | struct Point { X: i32, Y: i32 }
  |        ^^^^^
  = note: the matched value is of type `Point`
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern, a match arm with multiple or-patterns as shown, or multiple match arms
  |
8 ~         Point { X, Y } => {}
9 ~         Point { X: i32::MIN..=-1_i32, .. } | Point { X: 1_i32..=i32::MAX, .. } => todo!(),
  |
```

Since from the plugin point of view, `Point { X, Y }` is exhaustive here, it also produces false-positive warning for the following code

```rust
struct Point { X: i32, Y: i32 }

pub const X: i32 = 0;
pub const Y: i32 = 0;

fn func(foo: Point) {
    match foo {
        Point { X, Y } => {},
        _ => {}
    }
}
```

<img width="550" alt="Screenshot 2022-09-16 at 18 37 03" src="https://user-images.githubusercontent.com/2539310/190687505-067ed7f2-ed51-49d0-8174-ac117d7f26dd.png">
